### PR TITLE
CXX-595 Implement a basic from_json

### PR DESCRIPTION
**Pull request by acmorrow**

### CODE REVIEW

**Comment by acmorrow** [comment]
Probably needs a run through clang-format

File: src/bsoncxx/test/json.cpp [hunk 1]
```diff
@@ -0,0 +1,32 @@
+#include "catch.hpp"
+
+#include <bsoncxx/builder/stream/document.hpp>
+#include <bsoncxx/json.hpp>
+
+namespace {
+    const auto k_invalid_json = R"({])";
```
Comment (amidvidy) [inline src/bsoncxx/test/json.cpp]: constexpr? :)

Comment (acmorrow) [inline src/bsoncxx/test/json.cpp]: Done

**Comment by acmorrow** [comment]
Other thought is whether we want a version with more complex error reporting (like where we found the first wrong byte, etc.). I'm not sure it is worth the effort right now.

**Comment by amidvidy** [comment]
LGTM

**Comment by amidvidy** [comment]
merged in c9f1ad13

**Pull request closed without merging**